File: php/bob/bob.php
<?php
class Bob{
  public function respondTo($input){
    $bobSays = "";
    if(preg_match("/^\s*$/", $input)){
      $bobSays .= "Fine. Be that way!";
    }else if((preg_match("/[A-Z]{4,}/", $input))||(preg_match("/[A-Z]!/", $input))){
      $bobSays .= "Whoa, chill out!";
    }else if(preg_match("/\?\s*$/", $input)){
      $bobSays .= "Sure.";
    }else{
      $bobSays .="Whatever.";
    }
    return $bobSays;
  }
}


?>

File: ruby/bob/bob.rb
class Bob
  attr_reader :remark
  def initialize
  end
  def hey(remark='')
    case
    when remark.match(/([A-Z]{4,})|([A-Z]{2,}!)/)
      'Whoa, chill out!'
    when remark.match(/\?\z/)
      'Sure.'
    when remark.match(/\A\s*\z/)
      'Fine. Be that way!'
    else
      'Whatever.'
    end
  end
end
